runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/ComputationWorkExecutor.java

```diff
@@ -83,11 +83,11 @@ public final void executeWork(
    * cannot be reused.
    */
   public final void invalidate() {
-    context().invalidateCache();
     try {
+      context().invalidateCache();
```

Comment:
lets keep it outside and not suppress the exception. I'm not sure about the implications of suppressing the exception here, it is better to fail than operating in an inconsistent state.

Reply:
Discussed over chat.
Safe because:
- logic on invalidate cache will not run if work is null
- Only 1 thread holds a reference to ComputationWorkExecutor and the only way it can be reused by another thread is if it is released to ComputationState.  If an exception is thrown during processing, the ComputationWorkExecutor will not be reused and the reference freed.

Reply:
I think either is okay here, since even if we throw it the upper layer suppresses it. But in general it is better to not suppress unexpected exceptions and let the upper layer handle or re-throw it.

Reply:
Done